msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfigurationFactory.kt

```diff
@@ -0,0 +1,266 @@
+            val useRealAuthority = developerConfig?.useRealAuthority
+
+            if (dc != null) {
+                BuildValues.setDC(dc)
+            }
+
+            if (useRealAuthority != null) {
+                BuildValues.setUseRealAuthority(useRealAuthority)
+            }
+        }
+
+        /**
+         * This method loads the default Native Auth configuration
+         */
+        @WorkerThread
+        private fun loadDefaultNativeAuthConfiguration(context: Context): NativeAuthPublicClientApplicationConfiguration {
+            val methodTag = "$TAG:loadDefaultNativeAuthConfiguration"
+            Logger.verbose(methodTag, "Loading default native auth configuration")
+            // Load and return default Native Auth config
+            return loadConfiguration(context, R.raw.msal_native_auth_default_config)
+        }
+
+        /**
+         * Tweaked loadConfiguration to return [NativeAuthPublicClientApplicationConfiguration]
+         */
+        @WorkerThread
+        private fun loadConfiguration(
+            context: Context,
+            configResourceId: Int
+        ): NativeAuthPublicClientApplicationConfiguration {
+            val configStream = context.resources.openRawResource(configResourceId)
+            val useDefaultConfigResourceId = (configResourceId == R.raw.msal_native_auth_default_config)
+            return loadConfiguration(configStream, useDefaultConfigResourceId)
+        }
+
+        /**
+         * Tweaked loadConfiguration to return [NativeAuthPublicClientApplicationConfiguration]
+         */
+        @WorkerThread
+        private fun loadConfiguration(configFile: File): NativeAuthPublicClientApplicationConfiguration {
+            return try {
+                loadConfiguration(FileInputStream(configFile), false)
+            } catch (e: FileNotFoundException) {
+                throw java.lang.IllegalArgumentException("Provided configuration file path=" + configFile.getPath() + " not found.")
+            }
+        }
+
+        /**
+         * Tweaked loadConfiguration to serialize json into a [NativeAuthPublicClientApplicationConfiguration]
+         * object.
+         */
+        @WorkerThread
+        private fun loadConfiguration(
+            configStream: InputStream,
+            isDefaultConfiguration: Boolean
+        ): NativeAuthPublicClientApplicationConfiguration {
+            val methodTag = "$TAG:loadConfiguration"
+            val buffer: ByteArray
+            try {
+                buffer = ByteArray(configStream.available())
+                configStream.read(buffer)
+            } catch (e: IOException) {
+                if (isDefaultConfiguration) {
+                    throw IllegalStateException("Unable to open default native auth configuration file.", e)
+                } else {
+                    throw IllegalArgumentException("Unable to open provided native auth configuration file.", e)
+                }
+            } finally {
+                try {
```

Comment:
You may want to use `try-with-resources`. The Kotlin equivalent I believe is called `use`